cmd/commands/init_test.go

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the cpackget project. */
+
+package commands_test
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	errs "github.com/open-cmsis-pack/cpackget/cmd/errors"
+)
+
+// Tests for init command are placed here because there was something wrong
+// while putting them into a file init_test.go
+
+var (
+	pidxFilePath         = filepath.Join(testingDir, "SamplePublicIndex.pidx")
+	notFoundPidxFilePath = filepath.Join("path", "to", "index.pidx")
+)
+
+var initCmdTests = []TestCase{
+	{
+		name:        "test no parameter given",
+		args:        []string{"init"},
+		expectedErr: errors.New("accepts 1 arg(s), received 0"),
+	},
+	{
+		name:        "test help command",
+		args:        []string{"help", "init"},
+		expectedErr: nil,
+	},
+	{
+		name: "test create using an index.pidx",
+		args: []string{"init"},
+		setUpFunc: func(t *TestCase) {
+			server := NewServer()
+			t.args = append(t.args, server.URL()+"index.pidx")
+			server.AddRoute("index.pidx", []byte(`<?xml version="1.0" encoding="UTF-8" ?>
+<index schemaVersion="1.1.0" xs:noNamespaceSchemaLocation="PackIndex.xsd" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance">
+<vendor>TheVendor</vendor>
+<url>http://the.vendor/</url>
```

Comment:
should we be replacing http:// with https:// in our test files?

Reply:
Works for https://www.w3.org/2001/XMLSchema-instance, replaced the other URLs by "https"